docs/general/ddl/replication.rst

```diff
@@ -110,6 +95,86 @@ time.
     :ref:`CREATE TABLE: WITH clause <sql-create-table-number-of-replicas>`
 
 
+.. _ddl-replication-recovery:
+
+Shard recovery
+==============
+
+CrateDB :ref:`allocates <gloss-shard-allocation>` each primary and replica
+shard to a specific node. You can control this behavior by configuring the
+:ref:`allocation <conf_routing>` settings.
+
+If one or more nodes become unavailable (e.g., due to hardware failure or
+network issues), CrateDB will try to recover a replicated table by doing the
+following:
+
+.. rst-class:: open
+
+- For every lost primary shard, locate a replica and promote it to primary.
+
+  When CrateDB promotes a replica to primary, it can no longer function as a
+  replica, and so the total number of replicas decreases by one. Because each
+  primary requires a fixed :ref:`sql-create-table-number-of-replicas`, a new
+  replica has to be created (see next item).
+
+- For every primary with too few replicas (due to node loss or replica
+  promotion), use the primary shard to :ref:`recover <gloss-shard-recovery>`
+  the required number of replicas.
+
+Shard recovery is one of the features that allows CrateDB to provide continuous
+`availability`_ and `partition tolerance`_ in exchange for some
+:ref:`consistency trade-offs <concept-resiliency-consistency>`.
+
+.. SEEALSO::
+
+    `Wikipedia: CAP theorem`_
+
+.. _ddl-replication-underreplication:
+
+Underreplication
+================
+
+Having more replicas per primary and distributing shards as thinly as possible
+(i.e., fewer shards per node) can both increase chances of a :ref:`successful
+recovery <ddl-replication-recovery>` in the event of node loss.
+
+Although not ideal, a single node can hold multiple shards belonging to the
+same table. For example, suppose a table has more shards (primaries and
```

Comment:
Having multiple shards of the same table on one node is a common pattern and even recommended as it increases parallel execution per read query (multiple threads can be used).

Reply:
gotcha thx!